### PR TITLE
228 add medical type UI

### PR DESCRIPTION
**Pull request by Adrinels**
This PR add Medicine type used as theme for the future forum. It include : 
- a dataclass containing all different medicine type
- a UI for any professional to register in one or many of this category to be notified when a post on this theme is posted on the forum

For now the object is store in the preferences and always synchronize with the database

### CODE REVIEW

File: app/src/main/java/com/github/h3lp3rs/h3lp/storage/Storages.kt [hunk 1]
```diff
@@ -8,7 +8,7 @@ import java.lang.Boolean.parseBoolean
  * Enumeration of all useful (local) storages in H3LP
  */
 enum class Storages() {
-    USER_COOKIE(), MEDICAL_INFO(), SKILLS(),EMERGENCIES_RECEIVED();
+    USER_COOKIE(), MEDICAL_INFO(), SKILLS(),EMERGENCIES_RECEIVED(),FORUM_THEMES;
```
Comment (emduc) [inline app/src/main/java/com/github/h3lp3rs/h3lp/storage/Storages.kt:11]: Is that where we store the notification preferences ? The name "Theme" does not seem like the most appropriate

File: app/src/main/java/com/github/h3lp3rs/h3lp/MainPageActivity.kt [hunk 2]
```diff
@@ -354,8 +354,9 @@ class MainPageActivity : AppCompatActivity(), OnRequestPermissionsResultCallback
         if(item.itemId==R.id.button_tutorial){
             viewPresentation(findViewById<View>(android.R.id.content).rootView)
         }
```
Comment (emduc) [inline app/src/main/java/com/github/h3lp3rs/h3lp/MainPageActivity.kt:356]: Just for consistency I believe we use spaces between operators
```suggestion
        if(item.itemId == R.id.button_tutorial){
            viewPresentation(findViewById<View>(android.R.id.content).rootView)
        }
```

File: app/src/main/java/com/github/h3lp3rs/h3lp/MainPageActivity.kt [hunk 3]
```diff
@@ -354,8 +354,9 @@ class MainPageActivity : AppCompatActivity(), OnRequestPermissionsResultCallback
...
-        // else if(item.itemId==R.id.toolbar_settings){ }
-        // TODO ( Allow user to choose data he would like to keep private aka not sent to the database)
+        else if(item.itemId==R.id.toolbar_settings){
```
Comment (emduc) [inline app/src/main/java/com/github/h3lp3rs/h3lp/MainPageActivity.kt:357]:
```suggestion
        else if(item.itemId == R.id.toolbar_settings){
```

and here as well

**Review by emduc — APPROVED** [review]
Nice work ! Thanks for that. Not much to say overall since this is mainly new graphics. One suggestion would be to add a theme to the switches we use to turn notifications on and off so that they look like a bell :)

File: app/src/main/java/com/github/h3lp3rs/h3lp/professional/ProMainActivity.kt [hunk 4]
```diff
@@ -14,4 +16,9 @@ class ProMainActivity : AppCompatActivity() {
         setContentView(R.layout.activity_professional_main)
     }
 
+    fun gotoForumTheme(view: View) {
```
Comment (Lindt19) [inline app/src/main/java/com/github/h3lp3rs/h3lp/professional/ProMainActivity.kt:19]:
```suggestion
    fun goToForumTheme(view: View) {
```

File: app/src/main/java/com/github/h3lp3rs/h3lp/professional/ProfessionalTypeSelection.kt [hunk 5]
```diff
@@ -0,0 +1,103 @@
...
+    /**
+     * Load forum themes data
+     */
+    private fun loadData() {
+        val theme = storage.getObjectOrDefault(getString(R.string.forum_theme_key),
+            MedicalType::class.java, null) ?: return
+
+        toggleSwitch(theme.cardiology, R.id.cardioTxt)
+        toggleSwitch(theme.generalist, R.id.generalSwitch)
+        toggleSwitch(theme.gynecology, R.id.GynecologySwitch)
```
Comment (Lindt19) [inline app/src/main/java/com/github/h3lp3rs/h3lp/professional/ProfessionalTypeSelection.kt:64]:
```suggestion
        toggleSwitch(theme.gynecology, R.id.gynecologySwitch)
```

File: app/src/main/java/com/github/h3lp3rs/h3lp/professional/ProfessionalTypeSelection.kt [hunk 6]
```diff
@@ -0,0 +1,103 @@
...
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContentView(R.layout.activity_professional_type_selection)
+        storage = storageOf(FORUM_THEMES)
+        loadData()
+    }
+
+    /**
+     * When the user leaves the activity
```
Comment (Lindt19) [inline app/src/main/java/com/github/h3lp3rs/h3lp/professional/ProfessionalTypeSelection.kt:31]:
```suggestion
     * Saves data when the user leaves the activity
```

File: app/src/main/java/com/github/h3lp3rs/h3lp/professional/ProfessionalTypeSelection.kt [hunk 7]
```diff
@@ -0,0 +1,103 @@
...
+        )
+
+        storage.setObject(getString(R.string.forum_theme_key), MedicalType::class.java, theme)
+        storage.push()
+    }
+
+    /**
+     * return the boolean from a switch button
+     */
+    private fun getBooleanFromSwitch(id: Int):Boolean{
```
Comment (Lindt19) [inline app/src/main/java/com/github/h3lp3rs/h3lp/professional/ProfessionalTypeSelection.kt:100]:
```suggestion
    private fun getBooleanFromSwitch(id: Int): Boolean {
```

**Review by Lindt19 — APPROVED** [review]
I had just a few remarks about the naming and consistency details, but overall very clean implementation and good choice of the forum themes 👍🏻

**Pull request merged**